File: 0649-dota2-senate/0649-dota2-senate.py
class Solution:
    def predictPartyVictory(self, senate: str) -> str:
        rarr=[]
        darr=[]
        next=len(senate)
        for i in range(0, len(senate)):
            rarr.append(i) if senate[i]=="R" else darr.append(i)
    
        # console.log(rarr, darr)
   
        while len(rarr)>0 and len(darr)>0:
        
            r=rarr[0]
            d=darr[0]
            rarr.pop(0)
            darr.pop(0)
            if r<d:         
                rarr.append(next)
            else:
                darr.append(next)
                
            
            next+=1
           
        if len(rarr)==0:
            return "Dire"
        else:
            return "Radiant"

File: 0649-dota2-senate/0649-dota2-senate.js
/**
 * @param {string} senate
 * @return {string}
 */
var predictPartyVictory = function(senate) {
    let rarr=new Set(), darr=new Set()
    let ptr=0, next=senate.length
    for(let i=0;i<senate.length;i++){
        senate[i]==="R" ?  rarr.add(i) : darr.add(i)
    }
    // console.log(rarr, darr)
   
    while(rarr.size>0 && darr.size>0){
        
        let r=[...rarr][0], d=[...darr][0]
        if(r<d){
            rarr.delete(r)
            rarr.add(next++)
            // next++;
            darr.delete(d)
            
        }else{
            darr.delete(d)
            darr.add(next++)
            // next++
            rarr.delete(r)
        }
    }
    return rarr.size===0 ? "Dire" : "Radiant"
};